RxSwift/Traits/Single.swift

```diff
@@ -15,13 +15,9 @@ public enum SingleTrait { }
 /// Represents a push style sequence containing 1 element.
 public typealias Single<Element> = PrimitiveSequence<SingleTrait, Element>
 
-public enum SingleEvent<Element> {
-    /// One and only sequence element is produced. (underlying observable sequence emits: `.next(Element)`, `.completed`)
-    case success(Element)
-    
-    /// Sequence terminated with an error. (underlying observable sequence emits: `.error(Error)`)
-    case error(Swift.Error)
-}
+/// .success: One and only sequence element is produced. (underlying observable sequence emits: `.next(Element)`, `.completed`)
+/// .failure: Sequence terminated with an error. (underlying observable sequence emits: `.error(Error)`
+public typealias SingleEvent<Element> = Result<Element, Swift.Error>
```

Comment:
The other changes make sense but I'm not sure this one necessarily does. 
Yes, the type signatures do - but they mean different things. Not too sure, I'll let @kzaher decide on this.

Reply:
IMO, Result and SingleEvent have the same meaning here. I can't prove that but please just think about that a little bit more
Feel free to close that PR, if it's not suitable :)